### ISSUE TITLE
Unable to load any of {libcudnn_cnn.so.9.1.0, libcudnn_cnn.so.9.1, libcudnn_cnn.so.9, libcudnn_cnn.so} Invalid handle. Cannot load symbol cudnnCreateConvolutionDescriptor

### ISSUE THREAD

**Issue by sma380** [m-bain/whisperX#1100]
I don't know how to resolve this error. I have a modal app build that was deployed on jan, 15 2025 working flawlessly and have not touched the code since, however now it doesn't work after redeploying it. I have seen from git issues from a year ago that setting ctranslate2 to 4.4.0 will fix it but that doesn't work with the most up to date dependencies for faster whisper etc. Please let me know if anyone has an updated solution, I've been stuck on this for a while and could use the help, thanks!

**Comment by EarningsCall** [comment]
same issue

**Comment by Barabazs** [comment]
@sma380 @EarningsCall I updated a fix for this dependency issue. please try again with whisperx v3.3.2

**Comment by scmanjarrez** [comment]
This isn't working for me. Version 3.3.2
```
...
whisperx                 3.3.2
...
nvidia-cublas-cu12       12.4.5.8                                                                                                                                                                           
nvidia-cuda-cupti-cu12   12.4.127                                                                                                                                                                           
nvidia-cuda-nvrtc-cu12   12.4.127                                                                                                                                                                           
nvidia-cuda-runtime-cu12 12.4.127                                                                                                                                                                           
nvidia-cudnn-cu12        9.1.0.70                                                                                                                                                                           
nvidia-cufft-cu12        11.2.1.3                                                                                                                                                                           
nvidia-curand-cu12       10.3.5.147                                                                                                                                                                         
nvidia-cusolver-cu12     11.6.1.9                                                                                                                                                                           
nvidia-cusparse-cu12     12.3.1.170                                                                                                                                                                         
nvidia-cusparselt-cu12   0.6.2                                                                                                                                                                              
nvidia-nccl-cu12         2.21.5                                                                                                                                                                             
nvidia-nvjitlink-cu12    12.4.127                                                                                                                                                                           
nvidia-nvtx-cu12         12.4.127
...
onnxruntime              1.21.1 
```

**Comment by philipp** [comment]
Fix worked for me, too, just now.
Though the link above is from a while ago.
I used 9.14.0 from https://developer.nvidia.com/cudnn-downloads

**Comment by GUUser91** [comment]
not sure if this will work for other but these series of commands help me solve the problem. note i'm using rtx 5090.
```
pip install torch==2.8.0+cu129 torchaudio==2.8.0+cu129 torchvision==0.23.0+cu129 \
  --extra-index-url https://download.pytorch.org/whl/cu129
```
```
pip uninstall -y ctranslate2
```
```
pip install --no-cache-dir --force-reinstall "ctranslate2>=4.6.0"
```
```
pip install --no-cache-dir --force-reinstall "numpy==2.0.2"
```
```
pip uninstall -y nvidia-cudnn-cu12
```
```
rm -rf $VIRTUAL_ENV/lib/python3.11/site-packages/nvidia/cudnn
```
```
unset LD_LIBRARY_PATH
```
```

**Comment by meghavarshini** [comment]
Hello everyone. I am facing this issue as well. This is my first time working with whispers, so I may be running it incorrectly. I tried the solutions suggested in this comment thread, but was unable to fix the issue. 

I would really appreciate any recommendations or corrections:

command:
```
whisperx <audio_file> --compute_type float16 --language en \
--output_dir ./transcripts --output_format vtt --model large-v2 \
--align_model WAV2VEC2_ASR_LARGE_LV60K_960H --device cuda \
--highlight_words True --diarize_model ./models/pyannote-diarization \
--diarize --min_speakers 2 --beam_size 5
```

error:
``` 
Unable to load any of {libcudnn_cnn.so.9.1.0, libcudnn_cnn.so.9.1, libcudnn_cnn.so.9, libcudnn_cnn.so}
Invalid handle. Cannot load symbol cudnnCreateConvolutionDescriptor
```

Details:

```
Task: diarization and transcription
diarization  model: ./models/pyannote-diarization


NVIDIA-SMI 580.95.05              Driver Version: 580.95.05      CUDA Version: 13.0

software versions:
whisperx                 3.7.4
pytorch-lightning        2.5.5
pytorch-metric-learning  2.9.0
torch                    2.8.0+cu129
torch-audiomentations    0.12.0
torch_pitch_shift        1.2.5
torchaudio               2.8.0+cu129
torchmetrics             1.8.2
torchvision              0.23.0+cu129
nvidia-cublas-cu12       12.9.1.4
nvidia-cuda-cupti-cu12   12.9.79
nvidia-cuda-nvrtc-cu12   12.9.86
nvidia-cuda-runtime-cu12 12.9.79
nvidia-cufft-cu12        11.4.1.4
nvidia-cufile-cu12       1.14.1.1
nvidia-curand-cu12       10.3.10.19
nvidia-cusolver-cu12     11.7.5.82
nvidia-cusparse-cu12     12.5.10.65
nvidia-cusparselt-cu12   0.7.1
nvidia-nccl-cu12         2.27.3
nvidia-nvjitlink-cu12    12.9.86
nvidia-nvshmem-cu12      3.3.20
nvidia-nvtx-cu12         12.9.79

```

**Comment by dorienh** [comment]
I also have this issue. whisperx  3.7.4, and Cudnn 9.15

cat /usr/include/x86_64-linux-gnu/cudnn_version.h | grep CUDNN_MAJOR -A 2
#define CUDNN_MAJOR 9
#define CUDNN_MINOR 15
#define CUDNN_PATCHLEVEL 1
--
#define CUDNN_VERSION (CUDNN_MAJOR * 10000 + CUDNN_MINOR * 100 + CUDNN_PATCHLEVEL)


Package                 Version
----------------------- -----------
aiohappyeyeballs        2.6.1
aiohttp                 3.13.2
aiosignal               1.4.0
alembic                 1.17.1
annotated-types         0.7.0
antlr4-python3-runtime  4.9.3
anyio                   4.11.0
asteroid-filterbanks    0.4.0
attrs                   25.4.0
av                      15.1.0
certifi                 2025.11.12
cffi                    2.0.0
charset-normalizer      3.4.4
click                   8.3.0
coloredlogs             15.0.1
colorlog                6.10.1
contourpy               1.3.3
cryptography            46.0.3
ctranslate2             4.4.0
cycler                  0.12.1
deprecation             2.1.0
docopt                  0.6.2
dst                     0.0.0a0
einops                  0.8.1
faster-whisper          1.2.1
filelock                3.20.0
flatbuffers             25.9.23
fonttools               4.60.1
frozenlist              1.8.0
fsspec                  2025.10.0
h11                     0.16.0
h2                      4.3.0
hf-xet                  1.2.0
hpack                   4.1.0
httpcore                1.0.9
httpx                   0.28.1
huggingface-hub         0.36.0
humanfriendly           10.0
hyperframe              6.1.0
HyperPyYAML             1.2.2
idna                    3.11
Jinja2                  3.1.6
joblib                  1.5.2
julius                  0.2.7
kiwisolver              1.4.9
lightning               2.5.6
lightning-utilities     0.15.2
llvmlite                0.45.1
Mako                    1.3.10
markdown-it-py          4.0.0
MarkupSafe              3.0.3
matplotlib              3.10.7
mdurl                   0.1.2
more-itertools          10.8.0
mpmath                  1.3.0
multidict               6.7.0
networkx                3.5
nltk                    3.9.2
numba                   0.62.1
numpy                   2.0.2
omegaconf               2.3.0
onnxruntime             1.23.2
openai-whisper          20250625
optuna                  4.6.0
packaging               25.0
pandas                  2.2.3
pillow                  12.0.0
pip                     25.3
postgrest               2.24.0
primePy                 1.3
propcache               0.4.1
protobuf                6.33.1
psycopg2                2.9.11
pyannote.audio          3.4.0
pyannote.core           5.0.0
pyannote.database       5.1.3
pyannote.metrics        3.2.1
pyannote.pipeline       3.0.1
pycparser               2.23
pydantic                2.12.4
pydantic_core           2.41.5
Pygments                2.19.2
PyJWT                   2.10.1
pyparsing               3.2.5
python-dateutil         2.9.0.post0
pytorch-lightning       2.5.6
pytorch-metric-learning 2.9.0
pytz                    2025.2
PyYAML                  6.0.3
realtime                2.24.0
regex                   2025.11.3
requests                2.32.5
rich                    14.2.0
ruamel.yaml             0.18.16
ruamel.yaml.clib        0.2.14
safetensors             0.6.2
scikit-learn            1.7.2
scipy                   1.16.3
semver                  3.0.4
sentencepiece           0.2.1
setuptools              80.9.0
shellingham             1.5.4
six                     1.17.0
sniffio                 1.3.1
sortedcontainers        2.4.0
soundfile               0.13.1
speechbrain             1.0.3
SQLAlchemy              2.0.44
storage3                2.24.0
StrEnum                 0.4.15
supabase                2.24.0
supabase-auth           2.24.0
supabase-functions      2.24.0
sympy                   1.14.0
tabulate                0.9.0
tensorboardX            2.6.4
threadpoolctl           3.6.0
tiktoken                0.12.0
tokenizers              0.22.1
torch                   2.8.0
torch-audiomentations   0.12.0
torch_pitch_shift       1.2.5
torchaudio              2.8.0
torchmetrics            1.8.2
tqdm                    4.67.1
transformers            4.57.1
typer                   0.20.0
typing_extensions       4.15.0
typing-inspection       0.4.2
tzdata                  2025.2
urllib3                 2.5.0
websockets              15.0.1
wheel                   0.45.1
whisperx                3.7.4
yarl                    1.22.0

**Comment by dorienh** [comment]
Nevermind, I read that using Python 3.11 instead of 3.12 helped. As 3.12 doesn't use gpu built into python,
